File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int count = 0;
    public int goodNodes(TreeNode root) {
        if(root == null){
            return 0;
        }
        else{
            dfs(root, root.val);            
        }
        return count;
    }

    public void dfs(TreeNode root, int maxValue){
        if(root == null){
            return;
        }
        if(root.val >= maxValue){
            count++;
            maxValue = root.val;
        }
        dfs(root.left, maxValue);
        dfs(root.right, maxValue);
    }
}

File: Data Structures & Algorithms/delete-leaves-with-a-given-value/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def removeLeafNodes(self, root: Optional[TreeNode], target: int) -> Optional[TreeNode]:

        def dfs(node, prev_node, side):
            # recurse down

            if node.left:
                dfs(node.left, node, "left")
            if node.right:
                dfs(node.right, node, "right")
            if not node.left and not node.right:
                if node.val == target:
                    if side == "left":
                        prev_node.left = None
                    elif side == "right":
                        prev_node.right = None
                    else:
                        return None
            
            return node

        
        return dfs(root, None, "")


        